### ISSUE TITLE
Grid: editor is rendered outside of the grid after editRow on last item

### ISSUE THREAD

**Issue by sirbris** [vaadin/framework#10559]
Vaadin Framework version 8.2.1

When editRow is called on a grid with scrollbar and the last item is selected, the editor jumps (1 row height) below the grid.

Steps to reproduce:
- click "Add" till there's a scrollbar in the grid
- scroll down and select last item in the grid
- click "Add & Edit" 
-> Editor is placed outside the grid itself

```
public class Testvaadin8UI extends UI {

    @WebServlet(value = "/*", asyncSupported = true)
    @VaadinServletConfiguration(productionMode = false, ui = Testvaadin8UI.class, widgetset = "com.example.testvaadin8.widgetset.Testvaadin8Widgetset")
    public static class Servlet extends VaadinServlet {
    }

    public class TestBean {
        private String name = "name";
        private String value = "value";

        public String getName() {
            return name;
        }

        public void setName(String inName) {
            name = inName;
        }

        public String getValue() {
            return value;
        }

        public void setValue(String inValue) {
            value = inValue;
        }

    }

    private ArrayList<TestBean> items = new ArrayList<>();

    @Override
    protected void init(VaadinRequest request) {
        final VerticalLayout layout = new VerticalLayout();
        layout.setMargin(true);
        setContent(layout);

        Grid<TestBean> grid = new Grid<TestBean>(TestBean.class);
        grid.setDataProvider(new ListDataProvider<>(items));
        grid.setWidth("100%");
        grid.setHeight("400px");
        grid.getEditor().setEnabled(true);
        grid.getColumn("name").setEditorComponent(new TextField());
        grid.getColumn("value").setEditorComponent(new TextField());

        layout.addComponent(new Button("Add", event -> {
            items.add(new TestBean());
            grid.getDataProvider().refreshAll();
        }));
        layout.addComponent(new Button("Add & Edit", event -> {
            TestBean bean = new TestBean();
            items.add(bean);
            grid.getDataProvider().refreshAll();
            grid.select(bean);
            grid.getEditor().editRow(new ArrayList<TestBean>(((ListDataProvider<TestBean>) grid.getDataProvider()).getItems()).indexOf(bean));
        }));

        layout.addComponent(grid);
    }
```

**Comment by sirbris** [comment]
probably related: https://github.com/vaadin/framework/issues/10558

**Comment by sirbris** [comment]
fixed with 8.4.0
